src/post.js

```diff
@@ -1,296 +1,55 @@
+  return decompressor.decompress(src);
+};
 
-  function BaseDecompressor() {
-    this.dctxPtr = _LZ4JS_createDecompressionContext();
-    if (!this.dctxPtr) throw new Error('LZ4JS_createDecompressionContext');
-    LZ4JS_instances[this.dctxPtr] = this;
-  }
-
-  BaseDecompressor.prototype.decompress = function() {
-    _LZ4JS_decompress(this.dctxPtr) || this.cleanup();
+if (ENVIRONMENT_IS_NODE) {
+  lz4js['createCompressStream'] = function (options) {
+    return new CompressStream(options);
   };
 
-  BaseDecompressor.prototype.cleanup = function() {
-    _LZ4JS_freeDecompressionContext(this.dctxPtr);
-    delete LZ4JS_instances[this.dctxPtr];
-    if (this.$error) throw this.$error;
+  lz4js['createDecompressStream'] = function () {
+    return new DecompressStream();
   };
+}
 
-
-  ENVIRONMENT_IS_NODE && (function() {
-    var Transform = require('stream').Transform;
-    var inherits = require('util').inherits;
-
-    function CompressStream(options) {
-      BaseCompressor.call(this, options);
-      Transform.call(this, this.options);
-      this.initialized = false;
-      this.srcSize = 0;
-      this.dstSize = 0;
-      this.src = new Buffer(0);
-      this.dst = new Buffer(0);
-    }
-    inherits(CompressStream, Transform);
-    assign(CompressStream.prototype, BaseCompressor.prototype);
-
-
-    CompressStream.prototype.$read = function(srcPtr, size) {
-      HEAPU8.set(new Uint8Array(this.src.buffer, this.src.byteOffset, this.srcSize), srcPtr);
-      return this.srcSize;
-    };
-
-    CompressStream.prototype.$write = function(dstPtr, size) {
-      this.dst = new Buffer(HEAPU8.buffer).slice(dstPtr, dstPtr + size);
-      this.push(new Buffer(this.dst));
-    }
-
-    CompressStream.prototype['_transform'] = function(chunk, encoding, callback) {
-      try {
-        if (!this.initialized) {
-          this.compressBegin();
-          this.initialized = true;
-        }
-        var offset;
-
-        for (offset = 0; offset < chunk.length; offset += BUF_SIZE) {
-          this.srcSize = Math.min(chunk.length - offset, BUF_SIZE);
-          this.src = chunk.slice(offset, offset + this.srcSize);
-          this.compressUpdate();
-        }
-        callback();
-      } catch (error) {
-        callback(error);
-      }
-    };
-
-    CompressStream.prototype['_flush'] = function(callback) {
-      try {
-        this.compressEnd();
-        callback();
-      } catch (error) {
-        callback(error);
-      }
-    };
-
-    lz4['createCompressStream'] = function(options) {
-      return new CompressStream(options);
-    };
-
-    function DecompressStream() {
-      BaseDecompressor.call(this);
-      Transform.call(this, {});
-      this.srcSize = 0;
-      this.dstSize = 0;
-      this.src = new Buffer(0);
-      this.dst = new Buffer(0);
+// Replace Thenable interface with Promise for Asynchronous modules
+if (Module['then']) {
+  delete Module['then'];
+  Module['ready'] = new Promise((resolve, reject) => {
```

Comment:
`Module['ready']` is used inside latest emscripten so it doesn't seem that `ready` is working properly. Simply call `lz4init()` to get the initialized module.

Here is a part of the code generated by emscripten.

```js
// Set up the promise that indicates the Module is initialized
var readyPromiseResolve, readyPromiseReject;
Module['ready'] = new Promise(function(resolve, reject) {
  readyPromiseResolve = resolve;
  readyPromiseReject = reject;
});
    
// ...

  return lz4init.ready
}
```

Reply:
Refactored due to new module loading way.